server/src/main/java/io/druid/server/coordination/BatchDataSegmentAnnouncer.java

```diff
@@ -111,11 +103,26 @@ public void announceSegment(DataSegment segment) throws IOException
             if (availableZNode.getCount() >= config.getSegmentsPerNode()) {
               availableZNodes.remove(availableZNode);
             }
-
             done = true;
+          } else {
+            availableZNodes.remove(availableZNode);
```

Comment:
technically we don't need to remove the node, since it is possible for another segment announcement to fit in there, just not this specific one.